pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -164,7 +164,15 @@ def optimize_budget_distribution(
 
     if budget_ranges is None:
         budget_ranges = {
-            channel: (0, min(total_budget, parameters[channel][0]))
+            channel: (
+                0,
+                min(
+                    total_budget,
+                    9 * parameters[channel][1]
+                    if method == "michaelis-menten"
+                    else -np.log(0.1 / 1.9) / parameters[channel][1],
```

Comment:
similar here. 

If they have some meaning, it might be best to extract and put into a variable with name 😄

Reply:
else here implies `sigmoid`?

Reply:
Correct, implies sigmoid.

Regarding, why `-np.log(0.1 / 1.9)` because based on the function `(alpha - alpha * np.exp(-lam * x)) / (1 + np.exp(-lam * x))` the way to get the point on X when Y its at 90% is with this implementation. Example code below:
``` python
# Generate random parameters for alpha and lambda
np.random.seed(42)  # for reproducibility
alpha_random = np.random.uniform(5, 20)  # Random alpha between 5 and 20
lam_random = np.random.uniform(0.1, 1)   # Random lambda between 0.1 and 1

# Generate x values and corresponding function values
x_values = np.linspace(0, 50, 500)
y_values = custom_function(x_values, alpha_random, lam_random)

# Calculate the x value at which the function value is 90% of alpha
x_90_percent = -np.log(0.1 / 1.9) / lam_random

# Calculate the function value at this x value
y_90_percent = sigmoid(x_90_percent, alpha_random, lam_random)

# Plotting
plt.figure(figsize=(10, 6))
plt.plot(x_values, y_values, label='Custom Function Curve', linewidth=2)
plt.scatter([x_90_percent], [y_90_percent], color='red', zorder=5, label=f'90% of alpha (x={x_90_percent:.2f}, y={y_90_percent:.2f})')
plt.axhline(y=alpha_random, color='g', linestyle='--', label=f'alpha = {alpha_random:.2f}')
plt.axhline(y=y_90_percent, color='r', linestyle='--', label=f'90% of alpha = {y_90_percent:.2f}')
plt.xlabel('x')
plt.ylabel('f(x)')
plt.title('Custom Function')
plt.legend()
plt.grid(True)
plt.show()
```
![image](https://github.com/pymc-labs/pymc-marketing/assets/59846724/3277cb94-e609-4199-918e-9805db3cecbc)

Reply:
Nice, that's awesome. Thanks for the explanation. 
I'm for the allowing the user to set the budgets like in the most recent change. 
Maybe it is good to add reference in the docstring? 

Another idea (if there is some closed form of these functions) would be have some helper functions: 
- `get_michaelis_menten_quantile_budget(q=.9, lam=lam) -> 9 * lam`
- `get_sigmoid_quantile_budget(q=0.9, lam=lam) -> -np.log(0.1 / 1.9) / lam_random`

Then, defining them would might be clear to the user